bpf/lib/ip_addr.h

```diff
@@ -0,0 +1,11 @@
+#include "common.h"
+
+union ip_address {
```

Comment:
To be honest I don't have a specific goal in making this change. I was looking for opportunities to contribute, and issue #23657 seemed like a good option. I interpreted that issue as wanting the current change, but I definitely could have misunderstood. What was the intention of that issue?  I see that @pchaigno created that issue, and liked this comment so maybe he can weigh in as well.

Reply:
That makes sense to me. Maybe this isn’t such a good first issue then, the scope isn’t very clear.

Reply:
Can I help clarify anything? If you mean which unions to replace, then you can find them with `git grep union bpf/lib/`. They have different field names but you should be able to recognize the pattern.

Reply:
Sure, I guess I'm a little confused by your previous comment. You mentioned not wanting to make it a huge change, but then also mentioned you don't want to do it piece meal. So I'm not sure what exactly the scope is here, and how you'd like to space it out.

As you mentioned, a similar `union` is defined elsewhere in the codebase, for instance: 
1. https://github.com/cilium/cilium/blob/1.13.2/bpf/lib/common.h#L260-L266
2. https://github.com/cilium/cilium/blob/1.13.2/bpf/lib/common.h#L245-L251

However, if I were to wholesale replace all of these uses with a new combined `union`, wouldn't that require a pretty large change as each access would change? I could definitely be missing a nuance that would allow me to do this without a large change to each use.

Sorry in advance for the confusion, and thanks for the help!

Reply:
Totally agree about the verbosity issues with this approach, and I personally don't love magic macros as a rule 😆. I can do some research to see if there's another way.

Reply:
We already use macros to avoid code duplication in C structures. See `__notify_common_hdr` and `__notify_pktcap_hdr`. I think it's possible to do the same here without magic. The difficulty is probably around the different names these map structures use for their IPv4-or-IPv6 union field? How much code churn would we end up with if we renamed to the most commonly-used fields?